src/scratchtocatrobat/converter/mediaconverter.py

```diff
@@ -256,39 +240,51 @@ def convert(self, progress_bar = None):
                     # TODO: move test_converter.py to converter-python-package...
                     image_processing.save_editable_image_as_png_to_disk(editable_image, image_file_path, overwrite=True)
 
-            self._copy_media_file(scratch_md5_name, src_path, resource_info["dest_path"],
-                                  resource_info["media_type"])
+            current_basename, _ = os.path.splitext(current_filename)
+            self.file_rename_map[current_basename] = {}
+            self.file_rename_map[current_basename]["src_path"] = src_path
+            self.file_rename_map[current_basename]["dst_path"] = resource_info["dest_path"]
+            self.file_rename_map[current_basename]["media_type"] = resource_info["media_type"]
 
             if resource_info["media_type"] in { MediaType.UNCONVERTED_SVG, MediaType.UNCONVERTED_WAV }:
                 converted_media_files_to_be_removed.add(src_path)
 
-        self._update_file_names_of_converted_media_files()
+        self.rename_media_files_and_copy()
 
+        # delete converted png files -> only temporary saved
         for media_file_to_be_removed in converted_media_files_to_be_removed:
             os.remove(media_file_to_be_removed)
 
+    # rename the media files and copy them to the catrobat project directory
+    def rename_media_files_and_copy(self):
 
-    def _update_file_names_of_converted_media_files(self):
-        for (old_file_name, new_file_name) in self.renamed_files_map.iteritems():
-            look_data_or_sound_infos = filter(lambda info: info.fileName == old_file_name,
-                                      catrobat.media_objects_in(self.catrobat_program))
-            # assert len(look_data_or_sound_infos) > 0
+        def create_new_file_name(provided_file, index_helper, file_type):
+            _, ext = os.path.splitext(provided_file)
+            if file_type in {MediaType.UNCONVERTED_SVG, MediaType.IMAGE}:
+                return "img_#" + str(index_helper.assign_image_index()) + ext
+            else:
+                return "snd_#" + str(index_helper.assign_sound_index()) + ext
+
+        media_file_index_helper = helpers.MediaFileIndex()
+        for info in catrobat.media_objects_in(self.catrobat_program):
+            basename, _ = os.path.splitext(info.fileName)
+
+            # ignore these files, already correctly provided by the converter
```

Comment:
Just a thought: key sprites are now the only ones left being saved with a hash in their name. If those are not necessary, it would be cleaner to remove those hashes too. This could maybe be done in _key_filename_for in converter.py.